app/validators/password_is_strong_validator.rb

```diff
@@ -27,30 +25,101 @@ def validate_each(record, attribute, value)
     return if value.blank?
 
     if is_simple?(value)
-      record.errors[attribute] << 'must contain letters, numbers, and at least one special character'
+      record.errors.add attribute, 'must contain letters, numbers, and at least one special character'
     end
 
-    if contains_username?(record.username, value)
-      record.errors[attribute] << 'must not contain the username'
+    if contains_username?(record.username, value) && !record.username.blank?
```

Comment:
```suggestion
    if !record.username.blank? && contains_username?(record.username, value)
```

If the `record.username.blank?` then there is not reason to call `contains_username?`.